### PR TITLE
Adapt dataservice::read to ApiLookupClient.

### PR DESCRIPTION
**Pull request by kzvieriev**
We have new ApiLookupClient class in the core for Api requests. It
should replace internal ApiClientLookup for both read and write libs.
New lookup shared between repositories, so only one OlpClient used for
api requests, this will help to merge requests for same client.

Resolves: OLPEDGE-1697

Signed-off-by: Kostiantyn Zvieriev <ext-kostiantyn.zvieriev@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/tests/VolatileLayerClientImplTest.cpp [hunk 1]
```diff
@@ -862,6 +862,8 @@ TEST(VolatileLayerClientImplTest, PrefetchTilesCancellableFutureCancel) {
   auto cancellable = client.PrefetchTiles(
       read::PrefetchTilesRequest().WithTileKeys(tile_keys));
 
+  std::this_thread::sleep_for(std::chrono::milliseconds(300));
+
```
Comment (kzvieriev) [inline olp-cpp-sdk-dataservice-read/tests/VolatileLayerClientImplTest.cpp:866]: oops, will remove that

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/FetchOptions.h [hunk 2]
```diff
@@ -19,6 +19,8 @@
 
 #pragma once
 
+#include <olp/core/client/FetchOptions.h>
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/FetchOptions.h:22]: Why is this needed?

File: olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.h [hunk 3]
```diff
@@ -38,20 +39,21 @@ namespace repository {
 
 class CatalogRepository final {
  public:
-  static CatalogResponse GetCatalog(client::HRN catalog,
+  static CatalogResponse GetCatalog(client::ApiLookupClient client,
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.h:42]: I don't like this approach. Better move away from static methods and add CatalogClient constructor which takes in HRN and ApiLookupClient.
Same for all other repository classes.
On the long run we will have more and more parameters.

File: olp-cpp-sdk-core/src/client/ApiLookupClient.cpp [hunk 4]
```diff
@@ -28,6 +28,15 @@ ApiLookupClient::ApiLookupClient(const HRN& catalog,
...
 
+ApiLookupClient::ApiLookupClient(const ApiLookupClient& other) = default;
+
+ApiLookupClient& ApiLookupClient::operator=(const ApiLookupClient& other) =
+    default;
+
+ApiLookupClient::ApiLookupClient(ApiLookupClient&& other) = default;
+
+ApiLookupClient& ApiLookupClient::operator=(ApiLookupClient&& other) = default;
+
```
Comment (andescu) [inline olp-cpp-sdk-core/src/client/ApiLookupClient.cpp:39]: Remove `other` else we might face unused variable issues.
